File: algorithms/2315.count-asterisks/solution.ts
function countAsterisks(s: string): number {
  let count = 0;
  let inside = true;

  for (const c of s) {
    switch (c) {
      case "|":
        inside = !inside;
        break;
      case "*":
        if (inside) {
          count++;
        }
        break;
    }
  }

  return count;
}

File: algorithms/2315.count-asterisks/solution.rb
# @param {String} s
# @return {Integer}
def count_asterisks(s)
  count = 0
  inside = true

  s.chars.each do |c|
    case c
    when '|'
      inside = !inside
    when '*'
      count += 1 if inside
    end
  end

  count
end
